demos/stardist_demo/python/stardist_demo.py

```diff
@@ -0,0 +1,373 @@
+                else 'GPU_THROUGHPUT_AUTO'
+
+    return config_user_specified
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-i', '--input', required=True, help='Path to input image')
+    parser.add_argument('-m', '--model', required=True,
+                        help='Path to *.xml file of the model')
+    parser.add_argument('-d', '--device', default='CPU',
+                        help='Device to be utilized for deep learning inference')
+    parser.add_argument('-t', '--confidence_threshold', default=0.7, type=float,
+                        help='Probability threshold for detection')
+    parser.add_argument('--nms', dest='nms_threshold', default=0.5, type=float,
+                        help='NMS procedure threshold for IoU')
+    parser.add_argument('--tile_size', default=1024, type=int, help='Tile size')
+    parser.add_argument('--no_show', help="Optional. Don't show output.", action='store_true')
+
+    infer_args = parser.add_argument_group('Inference options')
+    infer_args.add_argument('-nireq', '--num_infer_requests', help='Optional. Number of infer requests',
+                            default=0, type=int)
+    infer_args.add_argument('-nstreams', '--num_streams',
+                            help='Optional. Number of streams to use for inference on the CPU or/and GPU in throughput '
+                                 'mode (for HETERO and MULTI device cases use format '
+                                 '<device1>:<nstreams1>,<device2>:<nstreams2> or just <nstreams>).',
+                            default='0', type=str)
+    infer_args.add_argument('-nthreads', '--num_threads', default=None, type=int,
+                            help='Optional. Number of threads to use for inference on CPU (including HETERO cases).')
+    args = parser.parse_args()
+
+    if args.input.endswith('.ndpi'):
+        img = OpenSlideImageReader(args.input)
+    else:
+        img = BFImageReader(args.input)
+
+    log.info('Initializing Inference Engine...')
+    ie = IECore()
+
+    plugin_config = get_plugin_configs(args.device, args.num_streams, args.num_threads)
+
+    log.info('Loading network...')
+
+    model = StarDistModel(ie, args)
+
+    detector_pipeline = AsyncPipeline(ie, model, plugin_config,
+                                      device=args.device, max_num_requests=args.num_infer_requests)
+
+    tile_id = (0, 0)
+    ready_tile_id = (0, 0)
+    num_tiles_x = ceil(img.width / args.tile_size)
+    num_tiles_y = ceil(img.height / args.tile_size)
+    log.info('Number of tiles: {}'.format(num_tiles_x * num_tiles_y))
+
+    if not args.no_show:
+        img_preview, preview_scale = img.get_preview(max_width=256)
+
+    def next_tile(tile_id):
+        x, y = tile_id
+        if x < num_tiles_x - 1:
+            return (x + 1, y)
+        elif y < num_tiles_y - 1:
+            return (0, y + 1)
+        else:
+            return None
+
+    start = time.time()
```

Comment:
Please reuse `PerformanceMetrics` https://github.com/openvinotoolkit/open_model_zoo/blob/7577edeaa83d8bc2d30f200cdc786fdfc1d4ddb1/demos/object_detection_demo/python/object_detection_demo.py#L248

It uses `perf_counter()` instead of `time()` to ensure time is steady, and along with reports of mean metrics after run, it reports a mean for last second on a screen.